### ISSUE TITLE
Singleton lists should be automatically converted to single element when command needs it

### ISSUE THREAD

**Issue by SUPERCILEX** [nushell/nushell#13674]
### Related problem

Examples:
```
sudo journalctl ...(which ringboard-x11).path
kill -s 18 (ps | where name =~ "ringboard-x11" | first).pid
```

In both of these cases I'm generating a list with a single element and you get a bunch of errors to deal with before figuring out how to convert the list to a single element. Also if I make a mistake and the list suddenly ends up with more than one element, now my script has silent bugs that are hard to debug (like killing the wrong pid for example).

### Describe the solution you'd like

If would be nice if the shell could transparently rewrite single item lists to just be that item if the command doesn't accept lists, and continue failing otherwise. Now in the common case of a singleton list, stuff will Just Work :tm: while you still get errors if the list isn't a singleton.

**Comment by sholderbach** [comment]
Propose a rule under which condition those coercions should happen and when not.

Because I can imagine there are a ton of situations where the coercion would be equally surprising. In its typed nature, Nushell tries to be quite explicit about which type a certain result has, then it is documented and potentially a bit verbose but predictable.

For external commands I think we should not change behavior based on whether there is an `extern` definition available or not. So you don't know if an extern accepts one or many arguments as Nushell.

**Comment by SUPERCILEX** [comment]
I think it would still be predictable. If a command argument accepts a `T` and I have a `List<T>`, then this conversion runs:
```
if list.len != 1:
  throw invalid arg exception
arg = list[0]
```

**Comment by Turtyo** [comment]
isn't it an error in the first to give a `List<T>` to something that requires `T` ?

**Comment by SUPERCILEX** [comment]
Right, this request is about performing an automatic type coercion.

**Comment by Turtyo** [comment]
Wouldn't it be more logical for a function to vectorize on a vec (ie do a `each`) instead of coercing `List<T>` to `T` in the unique case where `List<T>` is of length one ? This would keep it type coherent without having special case depending on if the list is of size 1 or not.
This is actually something already done in Nushell, I know some functions work on `List<T>`, I don't remember which though

**Comment by SUPERCILEX** [comment]
That's a neat idea! Though I still think it'd be useful block non singleton vecs in some cases (see the kill example). Maybe we can have both? Commands where vectorization makes sense do that and others get type coercion. Or maybe all this needs is a new command called `single` that does the conversion and errors when len!=1.
